jupyter_notebook_config.py

```diff
@@ -0,0 +1,11 @@
+import logging
+
+c.EventLog.allowed_schemas = [
+    'lab.jupyter.org/command-invocations'
+]
+
+def make_eventlog_sinks(eventlog):
+    # As an example, let's write these to local file
+    return [logging.FileHandler('events.log')]
+
+c.EventLog.handlers_maker = make_eventlog_sinks
```

Comment:
Hm. I think what I want here is an `event_sinks` trait, instead. This would be a list of "EventSink" objects that route events wherever my sinks live. For some external service to become a sink, it would need its own EventSink subclass (equivalent to the `logging.FileHandler` in this example). The `telemetry` package can provide base classes for different types of sinks. 

Making this trait a `Callable` confuses me. It offers a little **too much** freedom for routing to sinks. I wouldn't know how to begin without seeing a thorough example. I think I would rather be given a defined interface that has a clear entry point.

Reply:
Unless this is a constraint of Jupyter's config system... 

i.e. I don't know how these things get order—do we need to instantiate the logging handlers *after* the event log has been instantiated and that's why this needs to be a callable?

Reply:
Thinking about this more, I'm realizing handling (i.e. routing) events and consuming events (i.e. event sinks) are two separate pieces. Maybe this trait should maintain the name `handlers`, but become a `List` of handlers. We don't ask users to make the handlers. We provide base handlers the can choose from and extend.

Reply:
See response at https://github.com/jupyterhub/jupyterhub/pull/2542#discussion_r294029388

TLDR is: 1. Yes this is a constraint of traitlets, 2. Agree we should find a workaround, 3. We should avoid making a new EventSink interface unless it offers very clear advantages over the standard, well documented / implemented python logging interface.